algebird-core/src/main/scala/com/twitter/algebird/Field.scala

```diff
@@ -54,6 +54,14 @@ object FloatField extends Field[Float] {
     assertNotZero(r)
     l / r
   }
+  override def sum(t: TraversableOnce[Float]): Float = {
+    var sum = 0.0f
+    t.foreach(sum += _)
```

Comment:
The only catch is `TraversableOnce` has no way to use `while` without `toIterator`. In that case, building an iterator can be more expensive that `foreach`. Consider an array. If you need to write `foreach` for an array, you can do:
```scala
def foreach(f: T => Unit): Unit = {
  var idx = 0
  while(idx < arr.length) {
   f(arr(idx))
    idx += 1
  }
}
```
Which can be cheaper than wrapping with an Iterator.

I think we should assume foreach is fine until we really want to geek out on different data structure testing.